resources/win_postDeployAndCommitsToInsights.ps1

```diff
@@ -0,0 +1,325 @@
+}
+
+#Debug-Log "Response : $component_response"
+#Debug-Log "Response Type : $component_response.GetType().Name"
+#$component_response | Select-Object -ExpandProperty QueryResult | select-Object -ExpandProperty Results | Select-Object -ExpandProperty ObjectID
+
+$component_id = Get-ComponentObjectId $component_response
+if (-not $component_id) {
+    Debug-Log $component_response
+    Exit-WithError "Failed to find component in Insights, no component id found in response."
+}
+
+Info-Log "Component found in Insights. VSMComponent.ObjectId: $($component_id)"
+
+# Make a Deploy
+$deploy_response = Make-VsmDeploy $DEPLOY_IS_SUCCESSFUL $formatted_start_date $formatted_end_date $DEPLOY_MAIN_REVISION $component_id $DEPLOY_BUILD_ID
+if (-not $?) {
+    Info-Log "VSMDeploy creation failed with response:  $deploy_response"
+    Exit-WithError "Failed to create deploy in Insights"
+}
+
+Debug-Log "Response VSMDeploy Creation Error : $($deploy_response.CreateResult.Errors)"
+
+# Get Deploy ID
+$deploy_id = Get-DeployObjectId $deploy_response
+# Exit if we can't find the deploy id in the response (this could be for many reasons)
+if (-not $deploy_id) {
+    Error-Log "Failed to create deploy in Insights with response: $deploy_response"
+    Exit-WithError "Failed to create deploy in Insights, no deploy id found in response"
+}
+Info-Log "VSMDeploy created successfully with ObjectId: $deploy_id for Build ID: $DEPLOY_BUILD_ID"
+
+
+# Create the commit log we're going to loop over
+$log_file_path = "$GIT_REPO_LOC/commit_log"
+$git_commit_string = "$PREVIOUS_SUCCESS_BUILD_COMMIT..$CURRENT_BUILD_COMMIT"
+
+Debug-Log "Creating commit log for commits: $git_commit_string"
+# Debug-Log "Commit contents : $(git --git-dir="$GIT_REPO_LOC/.git" log --pretty=format:'%H %ad' --date=iso $git_commit_string)"
+rm $log_file_path -ErrorAction SilentlyContinue
+git --git-dir="$GIT_REPO_LOC/.git" log --pretty=format:'%H %ad' --date=iso $git_commit_string > $log_file_path
+
+Debug-Log "Commit log created successfully"
+
+# Loop over the commit log and make VSMChanges
+Get-Content $log_file_path | ForEach-Object {
+    $line = $_
+    $parts = $line -split ' '
+    $commit_id = $parts[0]
+    $timestamp = $parts[1..($parts.Length - 1)] -join ' '
+
+    Debug-Log "Parsing commit log line: $line"
+    # Continue to the next commit if we can't parse the commit id or timestamp
+    if (-not $commit_id -or -not $timestamp) {
+        Error-Log "Failed to parse commit log line: $line"
+        return
+    }
+
+    # Parse the date
+    $formatted_date = Parse-CommitLogTimestamp $timestamp
+
+    # Make the VSMChange
+    Debug-Log "Creating VSMChange for commit: $commit_id with revision time: $formatted_date"
+    $change_response = Make-VsmChange $commit_id $formatted_date $deploy_id
+    if (-not $?) {
+        Info-Log "VSMChange creation failed with response:  $change_response"
+        Error-Log "Failed to create VSMChange in Insights"
```

Comment:
Should we go ahead and exist if this failed? We exit below if we can't parse out its id